src/Protector.php

```diff
@@ -746,4 +742,72 @@ protected function isSanctumActive(): bool
     {
         return in_array('auth:sanctum', config('protector.routeMiddleware'));
     }
+
+    /**
+     * Returns the destination file path for the database dump.
+     *
+     * @param string $fileName
+     *
+     * @return string
+     */
+    protected function getDumpDestinationFilePath(string $fileName): string
+    {
+        if (preg_match('/filename="(?P<filename>.+)"/i', $fileName, $matches)) {
+            $destinationFileName = $matches['filename'];
+        }
+
+        return sprintf(
+            '%s%s%s',
+            $this->getConfigValueForKey('baseDirectory'),
+            DIRECTORY_SEPARATOR,
+            ($destinationFileName ?? 'remote_dump.sql')
+        );
+    }
+
+    /**
+     * Writes the remote database dump to a specified file path.
+     * Contents are retrieved in chunks from the provided stream.
+     * When the database dump is encrypted (indicated by whether Laravel Sanctum is enabled or not) those chunks will also be decrypted.
+     *
+     * @param StreamInterface $stream
+     * @param string          $destinationFilePath
+     * @param int             $chunkSize
+     * @param bool            $sanctumEnabled
+     *
+     * @return void
+     */
+    protected function writeDumpFile(StreamInterface $stream, string $destinationFilePath, int $chunkSize, bool $sanctumEnabled): void
+    {
+        $resource = StreamWrapper::getResource($stream);
+
+        $outputHandle = fopen($this->getDisk()->path($destinationFilePath), 'wb');
+
+        while (!feof($resource)) {
+            $chunk = stream_get_contents($resource, $chunkSize);
+
+            if ($sanctumEnabled) {
+                $chunk = $this->decryptString($chunk);
+            }
+
+            fwrite($outputHandle, $chunk);
+        }
+
+        fclose($outputHandle);
+    }
+
+    /**
+     * Determines encryption overhead.
```

Comment:
either expand this beyond the method name or leave this line out

Reply:
i removed it